### DISCUSSION TITLE
Tesla Gateway trying to Access Powerwall-Dashboard?

### DISCUSSION THREAD

**Discussion by s0schen** [jasonacox/Powerwall-Dashboard#258] [General]
Observation while perusing my Network System logs

It seems the Gateway is trying to access the VM that I am running the Powerwall-Dashboard on.   I have it on the IOT network so that it does not have access to the trusted  network.   Anyone know if this is normal behavior or the Gateway scanning the devices that access it?

![image](https://user-images.githubusercontent.com/4704443/234590739-d9631ff7-617c-4d79-96b7-2a1d3304a22a.png)

**Comment by jasonacox** [comment]
It would be interesting to see what port it is trying to access. These could be disconnected responses (requests from pypowerwall that timeout or TCP connection recycled but payload is still sent to ephemeral port). 

The pypowerwall proxy attempts to create persistent TCP connections to the Powerwall gateway. Subsequent queries will use that existing TCP connection.   Those connections can last for hours or days. However, most firewalls will eventually cause those TCP sessions to break after a while (some firewalls have a TTL setting you can adjust). When the firewall breaks those connections, there is a possibility that the Gateway still has packets to send "back" to the pypowerwall client.  Without a stateful session to track, the firewall assumes that is traffic that shouldn't happen. You would likely see errors in the pypowerwall logs when this occurs.
